tiledb/sm/fragment/fragment_info.cc

```diff
@@ -86,6 +86,10 @@ void FragmentInfo::append(const SingleFragmentInfo& fragment) {
   fragments_.emplace_back(fragment);
 }
 
+void FragmentInfo::pop_front() {
+  fragments_.erase(fragments_.begin());
```

Comment:
This is expensive (complexity linear in the size of the vector). Note that the number of fragments in some of our cases can be in order of millions. Since you are only popping from the front, perhaps you maintain a `start_pos_` member instead?

Reply:
Ah yes! I switched things around a bit to make this better.